pyinfra/operations/server.py

```diff
@@ -883,7 +883,7 @@ def user(
 
         # Users are often added by other operations (package installs), so check
         # for the user at runtime before adding.
-        yield "grep '{1}:' /etc/passwd || useradd {0} {1}".format(
+        yield "grep -E '^{1}:' /etc/passwd || useradd {0} {1}".format(
```

Comment:
```suggestion
        yield "grep '^{1}:' /etc/passwd || useradd {0} {1}".format(
```

As above ^

Reply:
This one too. And tests are fixed now, thanks for pointing me to them!